docs/design/proposals/late-initialization/late-initialization.md

```diff
@@ -0,0 +1,277 @@
+4. User experience for custom resource lifecycle 
+
+✅ → doable with no/some efforts
+
+❌ → not achievable
+
+### Solution 1
+
+In this solution, the status fields will include all the fields which are CreateOutput shape members. Current status 
+fields are the CreateOutput shape members which are not present in CreateInput shape (spec fields).
+
+During reconciliation, the Read(One|Many),Update request inputs will be created from desired state but giving precedence
+to status fields over spec in case of conflict.
+
+
+1. *Consistency with K8s conventions* ❌ 
+    1. In K8s convention, the spec fields are not duplicated into status fields. This will confuse k8s users when they 
+    describe their custom resources.
+2. *Consistency across ACK service controllers* ✅
+    1. There should be no complexity in following this approach across all ACK service controllers.
+3. *GitOps compatibility* ✅
+    1. There should be no complexity in implementing GitOps for ACK custom resources. 
+    2. Approach mentioned in low level design below should work.
+4. *User experience for custom resource lifecycle* ✅
+    1. Applying updates to custom resource can become confusing for the user.
+    2. To update custom resource spec, K8s user will have to check resource status because status will have most updated
+     value of all the fields.
+
+Other Cons :
+
+* The number of status fields will be very large and there will be a lot of duplication in custom resource description
+
+
+### Solution 2
+
+This solution builds on top of “Solution 1”, with the assumption that “Problem” mentioned above is not very common 
+amongst AWS APIs.
+
+In this solution, service teams can specify the spec fields which will be duplicated in both spec and status of custom
+resource using generator.yaml
+
+
+1. *Consistency with K8s conventions* ❌ 
+    1. In K8s convention, the spec fields are not duplicated into status fields. This will confuse k8s users when they 
+    describe their custom resources.
+2. *Consistency across ACK service controllers* ❌
+    1. Since all AWS APIs do not have this “Problem”, behavior across different service controllers will not seem consistent.
+3. *GitOps compatibility* ✅
+    1. There should be no complexity in implementing GitOps for ACK custom resources. 
+    2. Approach mentioned in low level design below should work.
+4. *User experience for custom resource lifecycle* ✅
+    1. Applying updates to custom resource can become confusing for the user.
+    2. To update custom resource spec, K8s user will have to check resource status because status will have most updated
+     value for some of the fields.
+
+
+
+### Solution 3 (Preferred Solution)
+
+In this solution, If a default value is added or existing value is updated as part of Create/Update API output, ACK service controller will update the spec/status of the custom resource accordingly. These updates will be persisted in custom resource’s Annotation to refer in future reconciliations.
```

Comment:
By `annotation` are you referring to `metadata.annotations` or another list in `status`, like `conditions`?